dot/sync/test_helpers.go

```diff
@@ -88,3 +91,43 @@ func BuildBlock(t *testing.T, instance runtime.Instance, parent *types.Header, e
 		Body:   body,
 	}
 }
+
+const (
+	defaultMinPeers     = 1
+	defaultMaxPeers     = 5
+	testTimeout         = time.Second * 5
+	defaultSlotDuration = time.Second * 6
+)
+
+func newTestChainSyncWithReadyBlocks(t *testing.T, readyBlocks *blockQueue) *chainSync {
+	header, err := types.NewHeader(common.NewHash([]byte{0}), trie.EmptyHash, trie.EmptyHash, 0,
+		types.NewDigest())
+	require.NoError(t, err)
+
+	bs := new(syncmocks.BlockState)
```

Comment:
Do you think you could use mockgen generated mocks instead? :thinking: 
The problem with the current version of `mockery` we use (see #2500) is that it doesn't assert expectations automatically unlike mockgen. The newer `mockery` generated mocks do (by using `t.Cleanup(func() { mock.AssertExpectations(t) }`) but that's not our case, and it really is a significant risk we add to our tests.

Reply:
Later.